app/src/main/java/org/mozilla/fenix/settings/creditcards/view/CreditCardEditorView.kt

```diff
@@ -76,12 +90,35 @@ class CreditCardEditorView(
                 interactor.onSaveCreditCard(fields)
             }
         }
+    }
 
-        card_number_input.text = state.cardNumber.toEditable()
-        name_on_card_input.text = state.billingName.toEditable()
+    private fun validateCreditCardInfo(): Boolean {
+        var validEntry = true
+
+        // validate credit card number
+        if (!card_number_input.text.toString().validateCreditCardNumber()) {
+            card_number_layout.error =
+                containerView.context.getString(R.string.credit_card_number_validation_error_message)
+            card_number_title.setTextColor(containerView.context.getColorFromAttr(R.attr.destructive))
+            validEntry = false
+        } else {
+            card_number_layout.error = null
+            card_number_title.setTextColor(containerView.context.getColorFromAttr(R.attr.primaryText))
+        }
 
-        bindExpiryMonthDropDown(state.expiryMonth)
-        bindExpiryYearDropDown(state.expiryYears)
+        // validate name on card
```

Comment:
Similar to below - can you check if Chrome allows an empty name? I assume it's possible to leave this empty. Let's also just file a separate issue for this so we can have UX guidance.